### PR TITLE
Next

### PR DESCRIPTION
**Pull request by ktalebian**
Add support for Plugins API

## Running Locally

1. Install [Twilio CLI](https://github.com/twilio/twilio-cli)
2. Clone this repo and switch to this branch. Run `npm i`
3. Run `twilio plugins:link /path/to/plugin-flex`

### CODE REVIEW

File: src/commands/flex/plugins/create-configuration.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  }
+
+  /**
+   * Main method
+   */
+  async doRun() {
+    const config = await this.doCreateConfiguration();
+
+    this._logger.newline();
+    this._logger.success(`🚀 Your configuration **v${config.version}** was successfully created`);
```
Comment (ahcai) [inline src/commands/flex/plugins/create-configuration.js:26]: 🚀

File: src/commands/flex/plugins/create-configuration.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    this._logger.newline();
+
+    this._logger.info('**Next Steps:**');
+    this._logger.info(
+      `Run {{$ twilio flex:plugins:release --version ${config.version}}} to enable your configuration on your flex instance`,
```
Comment (ahcai) [inline src/commands/flex/plugins/create-configuration.js:31]: nit: `Flex instance`

File: src/commands/flex/plugins/create-configuration.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  major: flags.boolean({
+    exclusive: ['patch', 'minor', 'version'],
+  }),
+  version: flags.string({
+    exclusive: ['patch', 'minor', 'major'],
+  }),
+  plugin: flags.string({
+    multiple: true,
+    required: true,
+    description: 'The plugin to install formatted as pluginName@version',
```
Comment (ahcai) [inline src/commands/flex/plugins/create-configuration.js:115]: `The plugin to install; formatted as pluginName@version`?

File: src/commands/flex/plugins/release.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    if (this.flags.version && !this.flags.plugin) {
+      await this.doCreateRelease(this.flags.version);
+    } else {
+      const config = await super.doCreateConfiguration();
+      await this.doCreateRelease(config.version);
+    }
+  }
+
+  async doCreateRelease(version) {
+    await progress(`Enabling configuration **v${version}**`, async () => this.createRelease(version), false);
```
Comment (ahcai) [inline src/commands/flex/plugins/release.js:30]: would using the release wording be better?
```
Releasing configuration  **v${version}**
```
and
```
🚀 Your configuration **v${version}** was successfully released
```

File: src/commands/flex/plugins/create-configuration.js (hunk 3 above)
Comment (ktalebian) [reply src/commands/flex/plugins/create-configuration.js:115]: The sentence after the `;` is not a complete sentence ;)

File: src/commands/flex/plugins/release.js (hunk 4 above)
Comment (ktalebian) [reply src/commands/flex/plugins/release.js:30]: Asking Ruma to review

Comment (rnairtwilio) [reply src/commands/flex/plugins/release.js:30]: We have been using the word enable in our docs to represent a configuration being released or being active on Flex. So I am okay with not using released.  Change the success message to "🚀 Configuration **v${version}** successfully enabled"

File: src/commands/flex/plugins/deploy.js [hunk 5]
```diff
@@ -1,39 +1,141 @@
...
+    this.scriptArgs = [];
+  }
+
+  /**
+   * Main method
+   */
+  async doRun() {
+    const args = ['--quiet', '--persist-terminal'];
+
+    await progress('Validating plugin deployment', async () => this.validateVersion(), false);
```
Comment (rnairtwilio) [inline src/commands/flex/plugins/deploy.js:25]: We need to include the log for mentioning the profile "Using profile dev" or "Using account details from profile: dev"

Comment (rnairtwilio) [inline src/commands/flex/plugins/deploy.js:25]: Validating input arguments?

File: src/commands/flex/plugins/create-configuration.js (hunk 3 above)
Comment (ahcai) [reply src/commands/flex/plugins/create-configuration.js:115]: true. 😂 comma?

File: src/commands/flex/plugins/deploy.js (hunk 5 above)
Comment (ktalebian) [reply src/commands/flex/plugins/deploy.js:25]: We may validate other things in the future. If the validation fails, the error message is descriptive. No need to mention what exactly we are validating; just that we are validating everything is good to go.

File: src/commands/flex/plugins/build.js [hunk 6]
```diff
@@ -18,7 +17,7 @@ class FlexPluginsBuild extends FlexPlugin {
 }
 
 FlexPluginsBuild.description = createDescription(
-  'Builds your Flex plugin and creates a JavaScript and sourcemap bundle',
+  'Builds Flex plugin and creates a JavaScript and sourcemap bundle',
```
Comment (ahcai) [inline src/commands/flex/plugins/build.js:20]: should this be `Build the Flex plugin`? some of the them are in the format `{action} Flex plugin` while others are `{action} the Flex plugin`

File: src/commands/flex/plugins/remove.js [hunk 7]
```diff
@@ -16,6 +16,6 @@ class FlexPluginsRemove extends FlexPlugin {
   }
 }
 
-FlexPluginsRemove.description = createDescription('Removes your Flex plugin', true);
+FlexPluginsRemove.description = createDescription('Removes the Flex plugin', true);
```
Comment (ahcai) [inline src/commands/flex/plugins/remove.js:19]: or should this one be `Removes Flex plugin`? just want to be consistent. personally like the `the` but @rnairtwilio can decide

**Pull request closed without merging**